osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,672 @@
+    switch (record.ExceptionInformation[0]) {
+    case 0:
+      errorMsg << " The memory could not be read.";
+      break;
+    case 1:
+      errorMsg << " The memory could not be written.";
+      break;
+    case 8:
+      errorMsg << " DEP access violation.";
+      break;
+    }
+    r["exception_message"] = errorMsg.str();
+  } else if ((record.ExceptionCode == EXCEPTION_IN_PAGE_ERROR) &&
+             (record.NumberParameters == 3)) {
+    std::ostringstream memAddrStr;
+    memAddrStr << "0x" << std::hex << record.ExceptionInformation[1];
+
+    std::ostringstream ntstatusStr;
+    ntstatusStr << "0x" << std::hex << record.ExceptionInformation[2];
+
+    errorMsg << "The instruction at " << exAddrStr.str()
+             << " referenced memory at " << memAddrStr.str() << "."
+             << " The required data was not placed into memory because of"
+             << " an I/O error status of " << ntstatusStr.str() << ".";
+    r["exception_message"] = errorMsg.str();
+  }
+
+  return Status();
+}
+
+Status logPIDAndTID(IDebugSystemObjects2* system, Row& r) {
+  unsigned long type = 0;
+  unsigned long procID = 0;
+  unsigned long threadID = 0;
+
+  if ((system->GetCurrentProcessSystemId(&procID) != S_OK) ||
+      (system->GetCurrentThreadSystemId(&threadID) != S_OK)) {
+    return Status(1);
+  }
+
+  r["pid"] = BIGINT(procID);
+  r["tid"] = BIGINT(threadID);
+  return Status();
+}
+
+Status logProcessUptime(IDebugSystemObjects2* system, Row& r) {
+  unsigned long uptime = 0;
+  if (system->GetCurrentProcessUpTime(&uptime) == S_OK) {
+    r["process_uptime"] = BIGINT(uptime);
+    return Status();
+  }
+  return Status(1);
+}
+
+Status logDumpTime(IDebugControl5* control, Row& r) {
+  unsigned long epoch = 0;
+  if (control->GetCurrentTimeDate(&epoch) != S_OK) {
+    return Status(0);
+  }
+
+  std::time_t datetime = epoch;
+  struct tm gmt;
+  gmtime_s(&gmt, &datetime);
+  std::stringstream dumpTimestamp;
+  dumpTimestamp << std::put_time(&gmt, "%Y-%m-%d %H:%M:%S UTC");
+  r["datetime"] = dumpTimestamp.str();
+  return Status();
+}
+
+Status logOSVersion(IDebugControl5* control, Row& r) {
+  unsigned long platformID;
+  unsigned long majorVersion;
+  unsigned long minorVersion;
+  unsigned long buildNumber;
+
+  if (control->GetSystemVersionValues(
+          &platformID, &majorVersion, &minorVersion, nullptr, &buildNumber) ==
+      S_OK) {
+    r["major_version"] = INTEGER(majorVersion);
+    r["minor_version"] = INTEGER(minorVersion);
+    r["build_number"] = INTEGER(buildNumber);
+    return Status();
+  }
+  return Status(1);
+}
+
+Status logDumpType(IDebugControl5* control, Row& r) {
+  unsigned long flags = 0;
+  if (control->GetDumpFormatFlags(&flags) != S_OK) {
+    return Status(1);
+  }
+
+  std::ostringstream activeFlags;
+  bool firstString = true;
+  // Loop through MINIDUMP_TYPE flags and log the ones that are set
+  for (auto const& flag : kMinidumpTypeFlags) {
+    if (flags & flag.first) {
+      if (!firstString) {
```

Comment:
This way of doing things is alright, but a more concise and clear way might be adding any found flags to a vector or set, and then `join`ing them at the end with a coma, something like the following perhaps?
```
  std::vector<std::string> activeFlags;
  for (auto const& flag : kMinidumpTypeFlags) {
    if (flags & flag.first) {
      activeFlags.push_back(flag.second);
    }
  }
  r["type"] = osquery::join(activeFlags, ",");
  return Status();
```